src/types.rs

```diff
@@ -139,15 +142,28 @@ impl Request {
             method: req.method().clone(),
             params: HashMap::new(),
             body,
+            path: req.path().to_string(),
+            host: req.connection_info().host().to_string(),
+            protocol: req.connection_info().scheme().to_string(),
         }
     }
 
     pub fn to_hashmap(&self, py: Python<'_>) -> Result<HashMap<&str, Py<PyAny>>> {
         let mut result = HashMap::new();
+
         result.insert("params", self.params.to_object(py));
         result.insert("queries", self.queries.to_object(py));
         result.insert("headers", self.headers.to_object(py));
         result.insert("body", self.body.to_object(py));
+        let method = PyString::new(py, self.method.as_ref()).to_object(py);
+        result.insert("method", method);
+        let path = PyString::new(py, self.path.as_ref()).to_object(py);
+        result.insert("path", path);
+        let host = PyString::new(py, self.host.as_ref()).to_object(py);
+        result.insert("host", host);
+        let protocol = PyString::new(py, self.protocol.as_ref()).to_object(py);
+        result.insert("protocol", protocol);
+
```

Comment:
One suggestion 

We can optimise this dictionary creation using `intern` strings.

Also, can you please rebase with the latest main to get the benchmarks?

Reply:
I've updated code. Can you check it, pls.